keps/sig-api-machinery/4192-svm-in-tree/README.md

```diff
@@ -121,118 +120,93 @@ As an end user using encryption at rest, whenever the key change is detected we
-      Items []StorageVersionMigration `json:"items" listType:"map" listMapKey:"type" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,2,rep,name=items"`
-    }
-    ```
-
-- APIs in-tree will be _converted to `built-in types`_ from CRD.
+The following API of the `storageversionmigrations.migration.k8s.io` API group is
+based on the [original out-of-tree API types](https://github.com/kubernetes-sigs/kube-storage-version-migrator/blob/60dee538334c2366994c2323c0db5db8ab4d2838/pkg/apis/migration/v1alpha1/types.go).
+
+```go
+// StorageVersionMigration represents a migration of stored data to the latest
+// storage version.
+type StorageVersionMigration struct {
+  metav1.TypeMeta `json:",inline"`
+  // Standard object metadata.
+  // More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+  // +optional
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+  // Specification of the migration.
+  // +optional
+  Spec StorageVersionMigrationSpec `json:"spec,omitempty"`
+  // Status of the migration.
+  // +optional
+  Status StorageVersionMigrationStatus `json:"status,omitempty"`
+}
+
+// Spec of the storage version migration.
+type StorageVersionMigrationSpec struct {
+  // The resource that is being migrated. The migrator sends requests to
+  // the endpoint serving the resource.
+  // Immutable.
+  Resource GroupVersionResource `json:"resource"`
+}
+
+// The names of the group, the version, and the resource.
+type GroupVersionResource struct {
+  // The name of the group.
+  Group string `json:"group,omitempty"`
+  // The name of the version.
+  Version string `json:"version,omitempty"`
+  // The name of the resource.
+  Resource string `json:"resource,omitempty"`
+}
+
+type MigrationConditionType string
+
+const (
+  // Indicates that the migration is running.
+  MigrationRunning MigrationConditionType = "Running"
+  // Indicates that the migration has completed successfully.
+  MigrationSucceeded MigrationConditionType = "Succeeded"
+  // Indicates that the migration has failed.
+  MigrationFailed MigrationConditionType = "Failed"
+)
+
+// Status of the storage version migration.
+type StorageVersionMigrationStatus struct {
+  // The latest available observations of the migration's current state.
+  // +patchMergeKey=type
+  // +patchStrategy=merge
+  // +listType=map
+  // +listMapKey=type
+  // +optional
+  Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+  // ResourceVersion to compare with the GC cache for performing the migration.
+  // This is the current resource version of given group, version and resource when
+  // kube-controller-manager first observes this StorageVersionMigration resource.
+  ResourceVersion string `json:"resourceVersion,omitempty"`
+  // LastMigratedResource is used to pick up migration from where it left off in case of failure.
+  LastMigratedResource string `json:"lastMigratedResourceNameHash,omitempty"`
```

Comment:
Yeah I don't think we need this at all, it doesn't give us anything and adds a lot of overhead since we have to update this every time we update an object.